public/pages/DetectorsList/Components/ListControls/ListControls.tsx

```diff
@@ -14,31 +14,73 @@
  */
 
 import {
+  EuiComboBox,
+  EuiComboBoxOptionProps,
   EuiFieldSearch,
   EuiFlexGroup,
   EuiFlexItem,
   EuiPagination,
 } from '@elastic/eui';
 import React from 'react';
+import { ALL_DETECTOR_STATES, ALL_INDICES } from '../../utils/constants';
+import { getDetectorStateOptions } from '../../utils/helpers';
 
 interface ListControlsProps {
   activePage: number;
   pageCount: number;
   search: string;
-  onSearchChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  selectedDetectorState: string;
+  selectedIndex: string;
+  indexOptions: EuiComboBoxOptionProps[];
+  onDetectorStateChange: (options: EuiComboBoxOptionProps[]) => void;
+  onIndexChange: (options: EuiComboBoxOptionProps[]) => void;
+  onSearchDetectorChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onSearchIndexChange: (searchValue: string) => void;
   onPageClick: (pageNumber: number) => void;
 }
 export const ListControls = (props: ListControlsProps) => (
   <EuiFlexGroup>
-    <EuiFlexItem>
+    <EuiFlexItem grow={false} style={{ width: '50%' }}>
       <EuiFieldSearch
         fullWidth={true}
         value={props.search}
         placeholder="Search"
-        onChange={props.onSearchChange}
+        onChange={props.onSearchDetectorChange}
         data-test-subj="detectorListSearch"
       />
     </EuiFlexItem>
+    <EuiFlexItem>
+      <EuiComboBox
+        id="selectedDetectorStates"
+        placeholder="All detector states"
+        isClearable={true}
+        singleSelection={true}
+        async
+        options={getDetectorStateOptions()}
+        onChange={props.onDetectorStateChange}
+        selectedOptions={
+          props.selectedDetectorState == ALL_DETECTOR_STATES
+            ? []
+            : [{ label: props.selectedDetectorState }]
+        }
+      />
+    </EuiFlexItem>
+    <EuiFlexItem>
+      <EuiComboBox
+        id="selectedIndices"
+        placeholder="All indices"
+        singleSelection={true}
+        async
+        options={props.indexOptions}
+        onChange={props.onIndexChange}
+        onSearchChange={props.onSearchIndexChange}
+        selectedOptions={
+          props.selectedIndex == ALL_INDICES
+            ? []
+            : [{ label: props.selectedIndex }]
+        }
+      />
```

Comment:
haha, it surprises to me that implementation of the Indices box is pretty similar to mine. Can you move this to a separate component so that it can be reused?

Another thing: `singleSelection={true}` should make user can only select 1 option, but I guess we may allow them to select 10 indices at most, as per previous discussion: https://aws-uxdr.invisionapp.com/d/main#/console/19392952/405140824/comments/127541146

Reply:
Yes, I forgot about that! I was going to discuss with you about this. What do you think about me making a "public/pages/components" and adding this there?

Reply:
Actually, I wonder if it's necessary to move to a separate component. For me, almost all of the props set in the EuiComboBox I have to supply from List.tsx where the state is manipulated. If I made this reusable, the only reusable fields would be the `id`, `placeholder`, `singleSelection`, and `async` fields. I will at least set `singleSelection={false}`.

Reply:
I see. My initial thought is to have a common component like Indices Box which can be reused by multiple pages(like Dashboard), same as State Box. But it may make the implementation code more complicated. I am not sure if it is worth doing that way. @mihirsoni could you share your thoughts on this?

Reply:
Sorry for chiming in late,

I agree with @ohltyler  if it doesn't abstract any logic doesn't makes sense to make it Common. If we can abstract all the implemeation and use it at two place we could do it. Also sometimes this small redundancy helps .

Reply:
I see. Thanks for your comment!